### PR TITLE
Published map layer style select

### PR DESCRIPTION
**Pull request by data-ux**
User can allow layer style change on published map in the publisher tool for layer menu.

This selection is stored in a new conf flag `isStyleSelectable` for the LayerSelectionPlugin.

### CODE REVIEW

File: bundles/framework/publisher2/tools/LayerSelectionTool.js [hunk 1]
```diff
@@ -50,6 +58,7 @@ Oskari.clazz.define('Oskari.mapframework.publisher.tool.LayerSelectionTool',
                     pluginConfig.config.baseLayers = layerSelection.baseLayers;
                     pluginConfig.config.defaultBaseLayer = layerSelection.defaultBaseLayer;
                 }
+                pluginConfig.config.isStyleSelectable = me.__plugin.getStyleSelectable();
```
Comment (ZakarFin) [inline bundles/framework/publisher2/tools/LayerSelectionTool.js]: Shouldn't this come in the this.getPlugin().getConfig() call above?

File: bundles/mapping/mapmodule/plugin/layers/LayerSelectionPlugin.js [hunk 2]
```diff
@@ -37,6 +37,7 @@ Oskari.clazz.define('Oskari.mapframework.bundle.mapmodule.plugin.LayerSelectionP
             },
             buttonGroup: 'mobile-toolbar'
         };
+        me._styleSelectable = !!this.getConfig().isStyleSelectable;
```
Comment (ZakarFin) [inline bundles/mapping/mapmodule/plugin/layers/LayerSelectionPlugin.js]: What's the difference with getConfig() and the config that is received as parameter? BasicMapModulePlugin handles setting the config that's returned by getConfig()?

File: bundles/framework/publisher2/tools/LayerSelectionTool.js [hunk 3]
```diff
@@ -211,32 +230,43 @@ Oskari.clazz.define('Oskari.mapframework.publisher.tool.LayerSelectionTool',
      * @return {Boolean} true if layer must be preselect, other false
      */
         shouldPreselectLayer: function (id) {
+            const toolPluginMapfullConf = this._getToolPluginMapfullConf();
+            if (toolPluginMapfullConf) {
+                var isPluginConfig = !!((toolPluginMapfullConf && toolPluginMapfullConf.config &&
+                    toolPluginMapfullConf.config.baseLayers));
+
+                if (isPluginConfig) {
+                    var isFound = jQuery.inArray('' + id, toolPluginMapfullConf.config.baseLayers);
```
[the remark this answers is not in the document]
Comment (data-ux) [reply bundles/framework/publisher2/tools/LayerSelectionTool.js]: It's old code that was moved around, but yeah we can refactor it while at it.

File: bundles/framework/publisher2/tools/LayerSelectionTool.js (hunk 1 above)
Comment (data-ux) [reply bundles/framework/publisher2/tools/LayerSelectionTool.js]: `getConfig()` does not return the changes made in the publisher to the plugin config. I guess it returns the initial config given to the plugin. That's why `baseLayers` and `defaultBaseLayer` is written separately, too, on the above lines.

File: bundles/mapping/mapmodule/plugin/layers/LayerSelectionPlugin.js (hunk 2 above)
Comment (data-ux) [reply bundles/mapping/mapmodule/plugin/layers/LayerSelectionPlugin.js]: Yes, `config` param of constructor is assigned to instance variable `_config` by BasicMapModulePlugin. `getConfig()` reads that.

**Pull request merged**